File: Estudos/Dart/Sintaxe/Variaveis.dart
void main() {

  int valor = 1;
  double pi = 3.14;
  String nome = 'Rafael';
  bool verdadeiro = true;
  
  var numero = 1; // indentifica qual o tipo de variavel mas ele não pode mudar o tipo
  numero = 2;
  
  dynamic variavel = 1;
  variavel = 'Um';
  
  print(valor);
  print(pi);
  print(nome);
  print(verdadeiro);
  print(numero);
  print(variavel);

  
  // Concatenação
  int codigo = 3;
  
  print('Código é $codigo ou ${codigo * 2}');
  
  print('Codigo ' + codigo.toString() + ' e tambem ' + 
        (codigo * 2 ).toString());
  
}

File: Estudos/TypeScript/Sintaxe/Variaveis.ts
    let nome: String = "Rafael";
    let idade: number = 16;
    let ativo: boolean = true;

    // : any  =  aceitar qualquer tipo
    let valor: any = "Qual quer coisa";
    valor = 10;
    valor = true;   

    // : unknown
    let dado: unknown = "texto";

    let vazio: null = null; // : null = vazio
    let indefinido: undefined = undefined; // : undefined = indefinido

    // const : não muda igual JS
    const imutavel: String = "imutavel";


    // Union Types
    let id: number | String;

    id = 10;
    id = "abc1234";

    // literal types = valores fixos
    let status1: "online" | "offline";
    status1 = "online";

    //enum = valores possiveis
    enum Status {
        ONLINE,
        OFFLINE,
        OCUPADO
    }
    
    let usuarioStatus: Status = Status.ONLINE;

    // Concatenação:
    console.log("Nome:", nome);
    console.log("Minha idade é " + idade);
    console.log(`Meu nome é ${nome} e tenho ${idade} anos.`);



